account_customer_wallet/models/res_partner.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2022 Coop IT Easy SCRL fs
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    customer_wallet_account_id = fields.Many2one(
+        comodel_name="account.account",
+        compute="_compute_customer_wallet_account_id",
+    )
+    # TODO: Is Monetary the correct field? What is behaviour with multiple
+    # currencies?
+    customer_wallet_balance = fields.Monetary(
+        string="Customer Wallet Balance",
+        currency_field="company_currency_id",
+        compute="_compute_customer_wallet_balance",
+        readonly=True,
+    )
+    company_currency_id = fields.Many2one(
+        comodel_name="res.currency",
+        related="company_id.currency_id",
+        string="Company Currency",
+        help="Utility field to express amount currency",
+        readonly=True,
+        store=True,
+    )
+
+    @api.depends("company_id.customer_wallet_account_id")
+    def _compute_customer_wallet_account_id(self):
+        for partner in self:
+            company = partner.company_id
+            partner.customer_wallet_account_id = company.customer_wallet_account_id
+
+    @api.depends(
+        "customer_wallet_account_id",
+        "customer_wallet_account_id.move_line_ids",
+        "customer_wallet_account_id.move_line_ids.balance",
+    )
+    def _compute_customer_wallet_balance(self):
```

Comment:
I think we could reuse same algorithm as in the ``_invoice_total`` function : 
https://github.com/odoo/odoo/blob/12.0/addons/account/models/partner.py#L264

To : 
- handle multicurrency. (I think that this point is optional, and can assume in the roadmap that the module works only in mono currency, but i think that one of your customer works in multi currency in switzerland...)
- handle parent / child computation.

Reply:
I don't think the linked algorithm handles multicurrency, unless I'm mistaken and the database does some conversions.

Reply:
you're right. That case, I think we should assume that the wallet is allways expressed in the company currency, and make conversion, if required. (but we can only raise an unimplementedError, if it's not in the scope of this PR).